### PR TITLE
feat: add snapshot cheatcodes

### PR DESCRIPTION
**Pull request by aon**
# What :computer: 
* Adds `vm.snapshot` and `vm.revertTo` cheatcodes

# Why :hand:
* To further support foundry cheatcodes on zksync.

# Evidence :camera:
image

### CODE REVIEW

File: crates/era-cheatcodes/tests/src/cheatcodes/Snapshot.t.sol [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            num,
+            "snapshot revert for block.number unsuccessful"
+        );
+        assertEq(
+            block.timestamp,
+            time,
+            "snapshot revert for block.timestamp unsuccessful"
+        );
+
+        // assertEq(
```
Comment (nbaztec) [inline crates/era-cheatcodes/tests/src/cheatcodes/Snapshot.t.sol:101]: commented code

File: crates/era-cheatcodes/tests/src/cheatcodes/Snapshot.t.sol [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        require(success, "warp failed");
+        assertEq(block.timestamp, 1337);
+
+        (success, ) = Constants.CHEATCODE_ADDRESS.call(
+            abi.encodeWithSignature("roll(uint256)", 99)
+        );
+        require(success, "roll failed");
+        assertEq(block.number, 99);
+
+        // vm.prevrandao(bytes32(uint256(123)));
```
Comment (nbaztec) [inline crates/era-cheatcodes/tests/src/cheatcodes/Snapshot.t.sol:82]: commented code

File: crates/era-cheatcodes/tests/src/cheatcodes/Snapshot.t.sol [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        console.log("store values: ", store.slot0, store.slot1);
+
+        assertEq(store.slot0, 10, "snapshot revert for slot 0 unsuccessful");
+        assertEq(store.slot1, 20, "snapshot revert for slot 1 unsuccessful");
+    }
+
+    function testBlockValues() public {
+        uint256 num = block.number;
+        uint256 time = block.timestamp;
+        // uint256 prevrandao = block.prevrandao;
```
Comment (nbaztec) [inline crates/era-cheatcodes/tests/src/cheatcodes/Snapshot.t.sol:61]: commented code

File: crates/era-cheatcodes/src/utils.rs [hunk 4]
```diff
@@ -31,3 +31,13 @@ impl ToH160 for Address {
...
+
+pub trait ToUint256_4 {
+    fn to_uint256_4(&self) -> Uint<256, 4>;
+}
+
+impl ToUint256_4 for U256 {
+    fn to_uint256_4(&self) -> Uint<256, 4> {
+        Uint::from_limbs(self.0)
+    }
+}
```
Comment (nbaztec) [inline crates/era-cheatcodes/src/utils.rs:43]: This might not be required `rU256::from_limbs_slice(&snapshot_id.0)` achieves the same.

**Review by nbaztec — APPROVED** [review]

File: crates/era-cheatcodes/src/utils.rs (hunk 4 above)
Comment (aon) [reply crates/era-cheatcodes/src/utils.rs]: Removed

**Pull request merged**